integrations/nifi/nifi-nar-bundles/nifi-hadoop-bundle/nifi-hadoop-processors/src/main/java/com/thinkbiganalytics/nifi/v2/hdfs/AbstractHadoopProcessor.java

```diff
@@ -199,6 +199,19 @@ public final void abstractOnScheduled(ProcessContext context) throws IOException
         }
     }
 
+    protected void resetFileSystem() throws IOException {
+        HdfsResources resources = hdfsResources.get();
+        Configuration conf = resources.configuration;
+        UserGroupInformation ugi = resources.userGroupInformation;
+        FileSystem fs = resources.fileSystem;
+        if (fs != null) {
+            fs.close();
+        }
+        fs = getFileSystemAsUser(conf,ugi);
```

Comment:
In non-Kerberos setting, `ugi` is null, resulting in` java.lang.NullPointerException`.